### PR TITLE
#166240992 User SignOut

### PR DESCRIPTION
**Pull request by Diama1**
#### What does this PR do?
Enables user to sign out from the system. 

#### Description of Task to be completed?
Implement the following endpoints:
- run Migration to create blacklist table. `sequelize db:migratre`
- `POST - /api/auth/login`
- `GET - /api/auth/signout`

#### How should this be manually tested?
Run the server and use Postman to test

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#166240992

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: src/api/controllers/auth.js [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+  static async SignOut(req, res) {
+    const { user: { id }, token } = req;
+    const droppedToken = await Blacklist.create({
+      userId: id,
+      token
+    });
+    if (droppedToken) {
+      res.json({
+        status: 200,
+        message: 'You are now signed in!',
```
Comment (Quantum-35) [inline src/api/controllers/auth.js]: Update db dropped message

File: src/middleware/auth.js [hunk 2]
```diff
@@ -1,52 +1,40 @@
...
-      return res.status(400).send({
-        status: 400,
-        error: 'Token is not provided'
-      });
+      return res.status(401).json({ status: 401, error: 'Token is missing' });
+    }
+
+    try {
+      const decoded = await authHelper.decodeToken(token);
+      try {
```
Comment (Quantum-35) [inline src/middleware/auth.js]: I don't see the essence of having a nested try.

**Review by Quantum-35 — CHANGES_REQUESTED** [review]

File: src/middleware/auth.js (hunk 2 above)
Comment (Diama1) [reply src/middleware/auth.js]: We are trying to catch database errors separately from token decoding errors and handle them separately.

File: src/index.js [hunk 3]
```diff
@@ -15,7 +16,9 @@ const app = express();
 
 globalMiddleware(app);
 app.use('/api', api);
-app.use('/', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
+app.get('/', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
```
Comment (deschantkn) [inline src/index.js]: Use `app.use` and place the swagger router below all other routes.

File: test/auth.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import chaiHttp from 'chai-http';
```
Comment (deschantkn) [inline test/auth.js]: I believe this file was renamed to `auth.test.js`. Could check if you don't already have it ? if you do then you can delete this file.

**Review by deschantkn — CHANGES_REQUESTED** [review]

**Review by Quantum-35 — APPROVED** [review]

**Review by EmyRukundo — APPROVED** [review]

**Pull request merged**